pkg/sql/control_schedules.go

```diff
@@ -57,19 +59,22 @@ func JobSchedulerEnv(execCfg *ExecutorConfig) scheduledjobs.JobSchedulerEnv {
 	return scheduledjobs.ProdJobSchedulerEnv
 }
 
-// loadSchedule loads schedule information.
+// loadSchedule loads schedule information as the node user.
 func loadSchedule(params runParams, scheduleID tree.Datum) (*jobs.ScheduledJob, error) {
 	env := JobSchedulerEnv(params.ExecCfg())
 	schedule := jobs.NewScheduledJob(env)
 
 	// Load schedule expression.  This is needed for resume command, but we
 	// also use this query to check for the schedule existence.
+	//
+	// Run the query as the node user since we perform our own privilege checks
+	// before using the returned schedule.
 	datums, cols, err := params.ExecCfg().InternalExecutor.QueryRowExWithCols(
 		params.ctx,
 		"load-schedule",
-		params.p.Txn(), sessiondata.InternalExecutorOverride{User: username.RootUserName()},
+		params.p.Txn(), sessiondata.InternalExecutorOverride{User: username.NodeUserName()},
```

Comment:
happy to support this, but why? :)

Reply:
My understanding is that `NodeUser` is what we should be using for intra-cluster traffic - https://github.com/cockroachdb/cockroach/blob/bf75f1df90d8a90a00eb52242dc791959c76de18/pkg/security/username/username.go#L67. It is the user used when a node is acting on its own behalf, rather than as an external user or an app running as root.

Reply:
Should it be running on its own behalf? Or should loadSchedule take a context and run as the user executing the query?

More concretely, I guess, can we imagine a world where we want a user to see/load some schedules but not others? (yes, I think?)

Reply:
> Or should loadSchedule take a context and run as the user executing the query?

This would be tricky because only `admin` is allowed to select against a system table. So a non-admin owner would get an error running `loadSchedule`. The scan of the system table is an internal implementation detail that expects other pieces of the logic to perform the appropriate privilege checks.

> can we imagine a world where we want a user to see/load some schedules but not others

Yes, I think so, but "load" is an internal implementation detail so I think the filtering needs to be in the logic that uses the loaded schedule, which is what we're doing here. We are already allowing users to only alter a limited set of schedules if they are non-admin, i.e. the schedules they own. Similarly if we want to limit what schedules a user sees we should be adding this privilege check to `SHOW SCHEDULES` and other user facing queries that allow for introspection.

Reply:
> expects other pieces of the logic to perform the appropriate privilege checks
Yeah, exactly this - I'd worry about a developer forgetting that. It's not obvious to me that somebody calling something called `loadSchedule()` will be executing node-level privileges under the hood. That said, if it's obvious to _you_ (i.e. common practice :) ) then I'm fine with it.